soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -6745,12 +6752,22 @@ s32 func_8083F570(Player* this, PlayState* play) {
 
             if (this->linearVelocity > 0.0f) {
                 this->actor.shape.rot.y = this->actor.wallYaw + 0x8000;
-                func_80832264(play, this, &gPlayerAnim_link_child_tunnel_end);
+                if (CVarGetInteger("gCrawlSpeed", 0) < 2) {
+                    func_80832264(play, this, &gPlayerAnim_link_child_tunnel_end);
+                } else {
+                    //animation when exiting a tunnel forward
+                    LinkAnimation_Change(play, &this->skelAnime, &gPlayerAnim_link_child_tunnel_end,
+                                         1.0f * ((CVarGetInteger("gCrawlSpeed", 0) + 1.0f) / 2.0f), 0.0f,
```

Comment:
Same here, this logically makes more sense as the if condition.

Reply:
This else is actually original code - going backward out of the tunnel. I was able to edit the play speed in-line, since they needed the LinkAnimation_Change function to play the animation in reverse.